virttest/utils_misc.py

```diff
@@ -2708,16 +2708,33 @@ def get_free_mem(session, os_type):
     :return string: freespace M-bytes
     """
     if os_type != "windows":
-        cmd = "grep 'MemFree:' /proc/meminfo"
-        output = session.cmd_output(cmd)
-        free = re.findall(r"\d+\s\w", output)[0]
+        free = "%s kB" % get_mem_info(session, 'MemFree')
     else:
         output = session.cmd_output("wmic OS get FreePhysicalMemory")
         free = "%sK" % re.findall("\d+", output)[0]
     free = float(normalize_data_size(free, order_magnitude="M"))
     return int(free)
 
 
+def get_mem_info(session=None, attr='MemTotal'):
```

Comment:
get_mem_info only applied to linux? And redundant code: "line 2734~2735" vs "line 2715~2716". I think an optimization is needed.

Reply:
@aliang123 Thank you for reviewing.

> get_mem_info only applied to linux? 

yes, It only applied for Linux, I haven't worked / have Windows env to implement for it.

> And redundant code: "line 2734-2735" with "line 2715-2716". I think an optimization is needed.

As `get_free_mem()` only get's free memory, `get_mem_info()` would help us to retrieve any memory information from /proc/meminfo as it is, ie in kB and `get_free_mem()` should not be disturbed as testcases are using it. so IMHO this 2 line will help us for better compatibility.

Reply:
@balamuruhans, got it. It looks good to me.
Acked-by: Aihua Liang<aliang@redhat.com>

Reply:
@aliang123 Thank you.